File: zip_deploy.py
import zipfile
import os

def zip_dir(path, zip_handler):
    for root, dirs, files in os.walk(path):
        for file in files:
            # Create a relative path for the file in the zip
            rel_path = os.path.relpath(os.path.join(root, file), path)
            # Use forward slashes for the zip entry name (VERY IMPORTANT for Linux AWS)
            zip_entry_name = rel_path.replace('\\', '/')
            zip_handler.write(os.path.join(root, file), zip_entry_name)

if __name__ == "__main__":
    zip_name = 'medvastr-prod-FINAL-SOLID.zip'
    source_dir = 'deploy_fixed'
    
    with zipfile.ZipFile(zip_name, 'w', zipfile.ZIP_DEFLATED) as zip_file:
        zip_dir(source_dir, zip_file)
    
    print(f"Successfully created {zip_name} with forward slashes.")

File: zip_deploy.js
const fs = require('fs');
const path = require('path');
const archiver = require('archiver');

const output = fs.createWriteStream(path.join(__dirname, 'medvastr-prod-FINAL-SOLID.zip'));
const archive = archiver('zip', {
    zlib: { level: 9 }
});

output.on('close', function () {
    console.log(archive.pointer() + ' total bytes');
    console.log('Successfully created medvastr-prod-FINAL-SOLID.zip with forward slashes.');
});

archive.on('error', function (err) {
    throw err;
});

archive.pipe(output);

const sourceDir = path.join(__dirname, 'deploy_fixed');

function addDirectory(dir, zipPath) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        const entryName = zipPath ? zipPath + '/' + file : file;

        if (stat.isDirectory()) {
            addDirectory(fullPath, entryName);
        } else {
            archive.file(fullPath, { name: entryName });
        }
    }
}

addDirectory(sourceDir, '');
archive.finalize();
